sklearn/metrics/_classification.py

```diff
@@ -321,17 +321,25 @@ def confusion_matrix(y_true, y_pred, *, labels=None, sample_weight=None,
                          "'all', None}")
 
     n_labels = labels.size
-    label_to_ind = {y: x for x, y in enumerate(labels)}
-    # convert yt, yp into index
-    y_pred = np.array([label_to_ind.get(x, n_labels + 1) for x in y_pred])
-    y_true = np.array([label_to_ind.get(x, n_labels + 1) for x in y_true])
+    # If labels are not consecutive integers starting from zero, then
+    # yt and yp must be converted into index form
```

Comment:
To make it easier for future maintainers:

```suggestion
    # y_true and y_pred must be converted into index form
```

Reply:
Fixed.